python/ray/util/sgd/v2/tests/test_session.py

```diff
@@ -106,6 +142,9 @@ def train_2():
     session.start()
     time.sleep(3)
 
+    session.pause_reporting()
+    session.get_next()
```

Comment:
is this `get_next` call necessary for the test to pass?

Reply:
Yeah this would be because I removed `session.continue_lock.release()` from `Session.finish()`,  so the training thread is stuck on `self.continue_lock.acquire()`. This is an example of the corner case mentioned in [this comment](https://github.com/ray-project/ray/pull/17632#discussion_r685631543), I can add a comment to the test to explain this!